nixos/lib/test-driver/test-driver.py

```diff
@@ -925,29 +874,16 @@ def join_all() -> None:
             machine.wait_for_shutdown()
 
 
-def test_script() -> None:
-    exec(os.environ["testScript"])
-
-
-def run_tests() -> None:
+def run_tests(interactive: bool = False) -> None:
     global machines
-    tests = os.environ.get("tests", None)
-    if tests is not None:
-        with log.nested("running the VM test script"):
-            try:
-                exec(tests, globals())
-            except Exception as e:
-                eprint("error: ")
-                traceback.print_exc()
-                sys.exit(1)
+    if interactive:
+        ptpython.repl.embed(globals(), locals())
     else:
-        ptpython.repl.embed(locals(), globals())
-
-    # TODO: Collect coverage data
-
-    for machine in machines:
-        if machine.is_up():
-            machine.execute("sync")
+        test_script()
+        # TODO: Collect coverage data
+        for machine in machines:
+            if machine.is_up():
+                machine.execute("sync")
```

Comment:
1. Does this actually do something, here? (machines have already finished the tests) -- might have been a legacy code smell.
2. Do we want to also call `join_all()`, here?

Reply:
i would not call join_all here. the test author might decide to just let all machines run after the test ran through and then be killed by the exiting test driver. 

i would not call it a "code smell" that the script makes all machines `sync` before quitting, but i guess we also don't need it anymore as we stopped doing all kinds of things after the tests.